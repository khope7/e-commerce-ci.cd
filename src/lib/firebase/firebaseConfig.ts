// firebaseConfig.ts
import { initializeApp } from "firebase/app";
import { getAuth } from "firebase/auth";
import type { Auth } from "firebase/auth";
import { getFirestore } from 'firebase/firestore';

// Pulling firebase api pull information from seperate env folder for security
const firebaseConfig = {
apiKey: process.env.VITE_APP_API_KEY,
authDomain: process.env.VITE_APP_AUTH_DOMAIN,
projectId: process.env.VITE_APP_PROJECT_ID,
storageBucket: process.env.VITE_APP_STORAGE_BUCKET,
messagingSenderId: process.env.VITE_APP_MESSAGING_SENDER_ID,
appId: process.env.VITE_APP_APP_ID
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);
const auth: Auth = getAuth(app);
const db = getFirestore(app);

export { auth };
export { db };