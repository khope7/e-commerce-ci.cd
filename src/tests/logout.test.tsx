import { render, fireEvent } from '@testing-library/react'
import Logout from '../pages/RegisterPages/Logout'
import { auth } from '../lib/firebase/firebaseConfig'

jest.mock('../lib/firebase/firebaseConfig', () => ({
auth: {
currentUser: { email: 'test@example.com' },
signOut: jest.fn(),
},
db: {},
}));


describe('Logout component', () => {
    test ('matches snapshot', () => {
        const {asFragment} = render(<Logout />);

        expect(asFragment()).toMatchSnapshot();
    })
    test ('Call signout when the button is clicked', () => {
        const {getByText} = render(<Logout/>);
        const button = getByText(/Logging Out!/i)
        fireEvent.click(button)
        expect(auth.signOut).toHaveBeenCalled()
    })
})
