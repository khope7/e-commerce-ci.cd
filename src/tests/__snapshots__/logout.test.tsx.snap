// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      Logging Out!
    </button>
  </div>
</DocumentFragment>
`;
